src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Formatting/Passes/HtmlFormattingPassBase.cs

```diff
@@ -48,6 +50,27 @@ public virtual async Task<ImmutableArray<TextChange>> ExecuteAsync(FormattingCon
         return changedText.GetTextChangesArray(originalText);
     }
 
+    private static ImmutableArray<TextChange> FilterIncomingChanges(FormattingContext context, ImmutableArray<TextChange> changes)
+    {
+        var syntaxTree = context.CodeDocument.GetSyntaxTree();
+
+        using var changesToKeep = new PooledArrayBuilder<TextChange>(capacity: changes.Length);
+
+        for (var i = 0; i < changes.Length; i++)
```

Comment:
Nit: A `foreach` loop is generally preferred, and it doesn't seem like `i` is doing anything important here.